src/core/data/allocator.cc

```diff
@@ -0,0 +1,55 @@
+/* Copyright 2022 NVIDIA Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include "core/data/allocator.h"
+
+namespace legate {
+
+IntraTaskAllocator::IntraTaskAllocator(Legion::Memory::Kind kind) : target_kind_(kind) {}
+
+IntraTaskAllocator::~IntraTaskAllocator()
+{
+  for (auto& pair : buffers_) { pair.second.destroy(); }
```

Comment:
Like I said in the previous comment, this will make the allocations escaped the scope unsafe to use. I think we should add an optional argument, say `scoped`, to the constructor that can control this behavior. When `scoped` is true, the destructor destroys all buffers in the destructor, otherwise, it lets them alive, allowing allocations to escape the scope. The latter is not necessarily unsafe, given that these buffers will be destroyed anyway when the task finishes. I think the default should be `scoped=true`.